op-node/rollup/derive/attributes.go

```diff
@@ -100,14 +99,23 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 			l2Parent, nextL2Time, eth.ToBlockID(l1Info), l1Info.Time()))
 	}
 
+	var upgradeTxs []hexutil.Bytes
+	if ba.cfg.IsEcotoneUpgradeDepositBlock(l2Parent.Time, nextL2Time) {
+		upgradeTxs, err = EcotoneNetworkUpgradeTransactions()
```

Comment:
This function is going to need to have the system config passed in so that it can migrate the scalar to the basefeeScalar

Reply:
Roberto implemented that base fee scalar update below